sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -633,7 +634,9 @@ Table 7. Member functions of the `command_graph` class (memory operations).
 [source,c++]
 ----
 using namespace ext::oneapi::experimental;
-node add_malloc_device(void*& data, size_t numBytes, const property_list& propList = {});
+template <typename T>
+node 
+add_malloc(T*& data, size_t count, sycl::usm::alloc kind, const property_list& propList = {});
```

Comment:
I think we may need exception behaviour for when the device doesn't support the requested USM kind, however this may need to be in finalize as it's at that point we know what device we're targetting

Reply:
I'm actually curious how we can return a device USM pointer here, when we don't know what device we're targetting at this point in time.

Reply:
That make sense, but what does it mean for the `data` return parameter? Would it be set to a placeholder for that USM allocation since we haven't done the allocation yet?  I think we'll need some wording to let the user know what valid usage of the value returned in `data` is.

```cpp
command_graph g;
float* foo = nullptr;
g.add_malloc<float>(foo, sycl::usm::alloc::device, n); // is foo set by this call to a placeholder?
auto node_i = g.add([&](sycl::handler& h) {
    h.parallel_for(n, [=](sycl::id<1> it){
      const size_t i = it[0];
      foo[i] = 3.14;
    });
}
auto exec_graph = g.finalize(q.get_context()); // Do the USM allocation
q.memset(foo, 42.0, sizeof(float);   // does foo still point to the placeholder?
sycl::free(foo, q.get_context());
```